serializer/src/main/java/serializer/peg/PegNode.java

```diff
@@ -320,6 +359,136 @@ public int hashCode() {
+         */
+        public PegNode getElse() {
+            return PegNode.idLookup(els).orElseThrow(IllegalStateException::new);
+        }
+    }
+
+    public static class ThetaNode extends OpNode {
+        static int _thetaId = 0;
+        private final int thetaId;
+        public final Integer init;
+        private boolean expand;   // indicates whether to expand during printing
+
+        private ThetaNode(final Integer idLit, final Integer init) {
+            super("theta", idLit, init);
+            thetaId = idLookup(idLit).orElseThrow(IllegalStateException::new)
+                      .asInteger().orElseThrow(IllegalStateException::new);
+            this.init = init;
+            this.expand = true;
+        }
+
+        @Override
+        public Optional<ThetaNode> asThetaNode() {
+            return Optional.of(this);
+        }
+
+        /**
+         * A {@code ThetaNode} may be identified with another {@code PegNode}. This method checks
+         * to see if it is and, if so, returns an {@code Optional.of(PegNode)} wrapping the identified
+         * node. Otherwise, returns empty.
+         *
+         * @return {@code Optional.of(node)} if this ThetaNode is identified with {@code node}; if {@code this}
+         * isn't identified with any nodes, return {@code Optional.empty()}
+         */
+        public Optional<PegNode> getIdentifiedNode() {
+            return Optional.ofNullable(thetaLookup.get(this.id))
+                    .flatMap(i -> Optional.ofNullable(idLookup.get(i)));
+        }
+
+        @Override
+        public <R, A> R accept(PegVisitor<R, A> visitor, A arg) {
+            return visitor.visit(this, arg);
+        }
+
+        @Override
+        public boolean isThetaNode() {
+            return true;
+        }
+
+        @Override
+        public String toDerefString() {
+            if (!expand) {
+                //return op + this.id;
+                return op;
+            }
+            expand = false;
+            final StringBuilder sb = new StringBuilder("(");
+            //sb.append(op + this.id);
+            sb.append(op);
+            sb.append(' ');
+            boolean added = false;
+            for (Integer child : children) {
+                if (added) {
+                    sb.append(' ');
+                }
+                added = true;
+                final PegNode p = idLookup.get(child);
+                if (p == null) {
+                    throw new IllegalStateException("OpNode " + op + " child index " + child + " not present");
+                }
+                sb.append(p.toDerefString());
+            }
+            expand = true;
+
+            return sb.append(")").toString();
+        }
+
+        /**
+         * @return the initializer node for this {@code ThetaNode} if it exists
+         * @throws IllegalStateException when the initializer id isn't associated with an {@code PegNode}
+         */
+        public PegNode getInitializer() {
```

Comment:
I actually deleted it because continuation is no longer a child. Continuation is now represented as the assigned theta. Previously it was a reference to a blank node. As long as we make it clear in the docs that getIdentifiedNode is returning essentially the continuation, we shouldn't need it